### ISSUE TITLE
Read more link is missing relevant a11y context 

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons#10090]
See: https://github.com/mozilla/addons-frontend/blob/4e7234fadbfc468375b9b7901842fd515529208f/src/ui/components/ShowMoreCard/index.js#L59

There a prefix of "Expand to" that's hidden but accessible, however this doesn't amount to enough of a context for an assistive tech user who might jump to that link. 

To be accessible this link needs to include a reference to what content is being expanded.
